micrometer-core/src/main/java/io/micrometer/core/instrument/cumulative/CumulativeDistributionSummary.java

```diff
@@ -69,12 +80,13 @@ public double max() {
     }
 
     private void updateMax(double amount) {
+        long amountLongBits = Double.doubleToLongBits(amount);
         while (true) {
-            double currentMax = max.get();
+            double currentMax = Double.longBitsToDouble(maxLongBits);
             if (currentMax >= amount) {
                 return;
             }
-            if (max.compareAndSet(currentMax, amount)) {
+            if (maxLongBitsUpdater.compareAndSet(this, maxLongBits, amountLongBits)) {
```

Comment:
Wow I learned a bit from this code. Very interesting! Would it be simpler to just use `java.util.concurrent.atomic.AtomicLong` to hold the bits? That would combine the updater and the `volatile` field?

Reply:
@checketts Ah, Thanks! `AtomicLong` is much better way. I'll update.

Reply:
@checketts I updated this PR in 0f54a0e760895263dc272c2cc82d869ce4c6efb0 as you suggested.